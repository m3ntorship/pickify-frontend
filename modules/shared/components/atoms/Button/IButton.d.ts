declare namespace IButton {
  export interface IProps {
    children?: string;
    disabled?: false | true;
    variant?: 'primary' | 'secondary' | 'text';
    size?: 'lg' | 'md' | 'sm';
    leftIcon?: false | true;
    rightIcon?: false | true;
    onlyIcon?: false | true;
    onClick?: () => void;
  }
}
export { IButton };
