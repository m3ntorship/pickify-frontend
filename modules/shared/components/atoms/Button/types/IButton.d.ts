import type * as EButton from './EButton';

declare namespace IButton {
  export interface IProps extends IHandlers {
    [property: string]: unknown;
    buttonText?: string | null;
    disabled?: boolean;
    variant: EButton.buttonVariantValues;
    size: EButton.buttonSizeValues;
    leftIcon?: boolean;
    rightIcon?: boolean;
    onlyIcon?: boolean;
    buttonType?: 'button' | 'reset' | 'submit';
  }

  export interface IHandlers {
    onClick?: () => void;
  }
}
export { IButton };
