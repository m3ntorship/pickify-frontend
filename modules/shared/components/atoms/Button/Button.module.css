/* ///////////////////////////////// */
.btn {
  @apply capitalize flex items-center justify-center  box-border m-0 focus:outline-none transition-all;
}
/* // the varient types of buttons   border-2 */
.btn-primary {
  @apply bg-primary rounded-full text-white hover:bg-primary-hover focus:ring-2 focus:ring-primary-dark m-0;
}

.btn-secondary {
  @apply rounded-full bg-white border border-primary text-primary hover:bg-grey-bg focus:ring-2 focus:ring-primary-dark  m-0;
}
.btn-tertiary {
  @apply rounded-full bg-white text-error m-0;
}

.btn-text {
  @apply bg-transparent text-accent hover:text-accent-hover border-0 focus:underline p-0;
}

.btn-like {
  @apply bg-primary rounded-full hover:ring-4 transition-all ring-primary ring-opacity-25 transform hover:scale-145;
}

.btn-dislike {
  @apply bg-error rounded-full hover:ring-4 transition-all ring-error ring-opacity-25 transform hover:scale-145;
}

.btn-delete {
  @apply rounded-full border border-grey hover:border-dark-grey;
}

/* // the size of the buttons  */
.btn-full-width {
  @apply text-sm font-medium w-full;
}

.btn-xlarge {
  @apply text-sm font-medium;
}

.btn-large {
  @apply text-md font-bold;
}

.btn-medium {
  @apply text-base font-medium;
}

.btn-normal {
  @apply text-sm font-medium;
}

.btn-small {
  @apply text-sm font-medium;
}

/* //////////// condition css //////////// */

/* the only of it is kind  -the responsive cancel button-  */
.btn-normal.btn-tertiary {
  @apply px-xll py-1.5;
}

.btn-full-width.btn-primary,
.btn-full-width.btn-primary,
.btn-full-width.btn-secondary,
.btn-full-width.btn-tertiary {
  @apply px-m py-1.5;
}

.btn-xlarge.btn-primary,
.btn-xlarge.btn-secondary,
.btn-xlarge.btn-tertiary {
  @apply px-4xvxl py-1.5;
}

.btn-large.btn-primary,
.btn-large.btn-secondary,
.btn-large.btn-tertiary,
.btn-large.btn-delete {
  @apply px-xl py-sx;
}
.btn-large.btn-primary.right-icon,
.btn-large.btn-secondary.right-icon {
  @apply pl-xl pr-lx  py-sx;
}
.btn-large.btn-primary.left-icon,
.btn-large.btn-secondary.left-icon {
  @apply pr-xl pl-l py-sx;
}

.btn-medium.btn-primary,
.btn-medium.btn-secondary,
.btn-medium.btn-tertiary {
  @apply px-l py-xsv text-base;
}
.btn-medium.btn-primary.right-icon,
.btn-medium.btn-secondary.right-icon {
  @apply pl-l pr-mx  py-xsv;
}
.btn-medium.btn-primary.left-icon,
.btn-medium.btn-secondary.left-icon {
  @apply pr-l pl-mx  py-xsv;
}

.btn-small.btn-primary,
.btn-small.btn-secondary {
  @apply px-m py-xxsv text-sm;
}
.btn-small.btn-primary.right-icon,
.btn-small.btn-secondary.right-icon {
  @apply pl-m pr-xsvv py-xxsv;
}
.btn-small.btn-primary.left-icon,
.btn-small.btn-secondary.left-icon,
.btn-small.btn-tertiary {
  @apply pr-m pl-xsvv py-xxsv;
}

/* ///////////////////only icon and it's padding  */

.btn-rounded.btn-large {
  @apply p-s;
}

.btn-rounded.btn-medium,
.btn-rounded.btn-small {
  @apply p-xs;
}

/* ////////////////////the proprties of the span holding the icon  */
.btn span,
.btn-rounded.btn-large span,
.btn-rounded.btn-medium span,
.btn-rounded.btn-small span {
  @apply flex items-center justify-center w-auto h-auto;
}

/* /////////////the arrow icon and its margins */

.right-icon.btn-large span,
.right-icon.btn-medium span {
  @apply ml-xs;
}
.right-icon.btn-small span {
  @apply ml-xxs;
}

.left-icon.btn-large span,
.left-icon.btn-medium span {
  @apply mr-xs;
}
.left-icon.btn-small span {
  @apply mr-xxs;
}
/* //////////////////////overwrite padding when we have an icon  */

/* ////////////////////the disabled buttons //////////// */
.btn[disabled] {
  @apply disabled:opacity-25 cursor-not-allowed;
}

.btn-primary[disabled] {
  @apply hover:bg-primary;
}

.btn-secondary[disabled] {
  @apply hover:bg-white;
}

/* Icons */

.btn-large span svg {
  @apply w-8 h-8;
}

.btn-medium span svg {
  @apply w-6 h-6;
}

.btn-small span svg {
  @apply w-4 h-4;
}

/* .btn span svg {
  @apply fill-current;
} */
