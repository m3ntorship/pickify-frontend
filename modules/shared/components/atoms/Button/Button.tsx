import React from 'react';
import classnames from 'classnames';
import type { FC, ReactElement } from 'react';
import type { IButton } from './types/IButton';
import styles from './Button.module.css';
import ArowDown from '../../icons/buttonArowDown.svg';

const Button: FC<IButton.IProps> = ({
  children = 'Button',
  disabled = false,
  variant = 'primary',
  size = 'medium',
  leftIcon = false,
  rightIcon = false,
  onlyIcon = false,
  buttonType,
  onClick,
}): ReactElement => {
  const btnClasses = classnames(
    styles.btn,
    {
      [styles['btn-primary']]: variant === 'primary',
      [styles['btn-secondary']]: variant === 'secondary',
      [styles['btn-tertiary']]: variant === 'tertiary',
      [styles['btn-text']]: variant === 'text',
    },
    {
      [styles['btn-xlarge']]: size === 'xlarge',
      [styles['btn-large']]: size === 'large',
      [styles['btn-medium']]: size === 'medium',
      [styles['btn-small']]: size === 'small',
    },
    {
      [styles['right-icon']]: rightIcon,
      [styles['left-icon']]: leftIcon,
    },
    { [styles['btn-rounded']]: onlyIcon },
  );

  return (
    <button
      /* eslint-disable react/button-has-type */
      type={buttonType}
      onClick={onClick}
      className={btnClasses}
      disabled={disabled}
    >
      {leftIcon && (
        <span>
          <ArowDown />
        </span>
      )}

      {onlyIcon ? (
        <span>
          <ArowDown />
        </span>
      ) : (
        children
      )}
      {rightIcon && (
        <span>
          <ArowDown />
        </span>
      )}
    </button>
  );
};

export default Button;
