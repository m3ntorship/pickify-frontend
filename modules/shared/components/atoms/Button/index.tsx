import type { FC, ReactElement } from 'react';
import React from 'react';
import classnames from 'classnames';
import type { IButton } from './IButton';

const Button: FC<IButton.IProps> = ({
  children = 'Button',
  disabled = false,
  variant = 'primary',
  size = 'md',
  leftIcon = false,
  rightIcon = false,
  onlyIcon = false,
  onClick,
}): ReactElement => {
  const buttonPaddingClassNames = classnames({
    'p-xs text-base font-medium': onlyIcon && size === 'md',
    'p-xs text-sm font-medium': onlyIcon && size === 'sm',
    'p-s text-md font-bold': onlyIcon && size === 'lg',
    'py-xs px-l text-base font-medium': !onlyIcon && size === 'md',
    'py-s px-m text-sm font-medium': !onlyIcon && size === 'sm',
    'py-m px-xl text-md font-bold': !onlyIcon && size === 'lg',
  });
  const iconSizeClassNames = classnames({
    '24': (size === 'md' && onlyIcon) || (size === 'lg' && !onlyIcon),
    '16':
      (size === 'sm' && onlyIcon) ||
      (size === 'md' && !onlyIcon) ||
      (size === 'sm' && !onlyIcon),
    '32': size === 'lg' && onlyIcon,
  });
  const buttonVariantClassNames = classnames({
    'text-primary bg-white hover:bg-grey-bg border border-primary focus:outline-none focus:ring-2 focus:ring-primary-dark ':
      variant === 'secondary',
    'bg-none text-accent hover:text-accent-hover focus:underline':
      variant === 'text',
    'bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-2 focus:ring-primary-dark':
      variant === 'primary',
  });
  const svgColorClassNames = classnames({
    'text-primary': variant === 'secondary',
    'text-accent': variant === 'text',
    'text-white': variant === 'primary',
  });
  return (
    <button
      type="button"
      onClick={onClick}
      className={`rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus
       ${buttonVariantClassNames}
       ${buttonPaddingClassNames}
       ${disabled ? 'opacity-25 pointer-events-none' : ''}`}
      disabled={disabled}
    >
      {rightIcon ? (
        <svg
          className={`${
            onlyIcon ? '' : 'mr-xxs'
          } inline-block fill-current ${svgColorClassNames}`}
          width={iconSizeClassNames}
          height={iconSizeClassNames}
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z" />
        </svg>
      ) : null}
      {/* {onlyIcon ? null : children} */}
      {onlyIcon ? (
        <svg
          className={`mx-xxs inline-block fill-current  ${svgColorClassNames} `}
          width={iconSizeClassNames}
          height={iconSizeClassNames}
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z" />
        </svg>
      ) : (
        children
      )}
      {leftIcon ? (
        <svg
          className={`${
            onlyIcon ? '' : 'ml-xxs'
          } inline-block fill-current  ${svgColorClassNames}`}
          width={iconSizeClassNames}
          height={iconSizeClassNames}
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z" />
        </svg>
      ) : null}
    </button>
  );
};

export default Button;
