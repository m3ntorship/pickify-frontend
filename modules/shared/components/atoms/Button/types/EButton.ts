export enum buttonVariantValues {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  QUATERNARY = 'quaternary',
  TEXT = 'text',
  LIKE = 'like',
  DISLIKE = 'dislike',
  DELETE = 'delete',
}
export enum buttonSizeValues {
  FULLWIDTH = 'fullWidth',
  XLARGE = 'xlarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  NORMAL = 'normal',
  SMALL = 'small',
}
