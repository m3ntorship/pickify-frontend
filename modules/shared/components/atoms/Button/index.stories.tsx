import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import Button from '.';
import { IButton } from './IButton';

export default {
  component: 'Atoms/Button',
  title: 'TS/Button',
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['lg', 'md', 'sm'],
      },
    },
    variant: {
      control: {
        type: 'inline-radio',
        options: ['primary', 'secondary', 'text'],
      },
    },
  },
};
const Template: Story<IButton.IProps> = (args): ReactElement => (
  <Button {...args} />
);

export const Default = Template.bind({});
Default.args = {
  size: IButton.buttonSizeValues.large,
  variant: IButton.buttonVariantValues.PRIMARY,
  disabled: false,
  leftIcon: false,
  rightIcon: false,
  onlyIcon: false,
  children: 'Button',
};
