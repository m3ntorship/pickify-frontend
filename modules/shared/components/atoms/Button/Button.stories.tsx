import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import Button from './Button';
import type { IButton } from './types/IButton';
import * as EButton from './types/EButton';
import ArrowDownIcon from '../../icons/arrowDown.svg';

export default {
  component: Button,
  title: 'atoms/Button',
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['xlarge', 'large', 'normal', 'medium', 'small'],
      },
    },
    variant: {
      control: {
        type: 'inline-radio',
        options: ['primary', 'secondary', 'tertiary', 'text'],
      },
    },
  },
};
const Template: Story<IButton.IProps> = (args): ReactElement => (
  <Button {...args}>
    <ArrowDownIcon />
  </Button>
);

export const Default = Template.bind({});
Default.args = {
  size: EButton.buttonSizeValues.MEDIUM,
  variant: EButton.buttonVariantValues.PRIMARY,
  disabled: false,
  leftIcon: false,
  rightIcon: false,
  onlyIcon: false,
  children: 'Button',
};
