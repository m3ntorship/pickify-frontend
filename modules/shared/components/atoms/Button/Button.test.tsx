import * as React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import Button from './Button';
import * as EButton from './types/EButton';
import ArrowDownIcon from '../../icons/arrowDown.svg';

describe('Button component', () => {
  it('should render (large) button when we apply (large)', () => {
    const tree = renderer
      .create(
        <Button
          size={EButton.buttonSizeValues.LARGE}
          variant={EButton.buttonVariantValues.PRIMARY}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (MEDIUM) button when we apply (MEDIUM)', () => {
    const tree = renderer
      .create(
        <Button
          size={EButton.buttonSizeValues.MEDIUM}
          variant={EButton.buttonVariantValues.PRIMARY}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (small) button when we apply (small)', () => {
    const tree = renderer
      .create(
        <Button
          size={EButton.buttonSizeValues.SMALL}
          variant={EButton.buttonVariantValues.PRIMARY}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (primary) button when we apply primary', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.PRIMARY}
          size={EButton.buttonSizeValues.MEDIUM}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (secondary) button when we apply (secondary)', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.SECONDARY}
          size={EButton.buttonSizeValues.MEDIUM}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (text) button when we apply text', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.TEXT}
          size={EButton.buttonSizeValues.MEDIUM}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (disapled) button when we apply disapled', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.PRIMARY}
          size={EButton.buttonSizeValues.MEDIUM}
          disabled
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (righticon) button when we apply righticon', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.PRIMARY}
          size={EButton.buttonSizeValues.MEDIUM}
          rightIcon
        >
          <ArrowDownIcon />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (lefticon) button when we apply lefticon', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.PRIMARY}
          size={EButton.buttonSizeValues.MEDIUM}
          leftIcon
        >
          <ArrowDownIcon />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render (onlyicon)  button when we apply nothing', () => {
    const tree = renderer
      .create(
        <Button
          variant={EButton.buttonVariantValues.PRIMARY}
          size={EButton.buttonSizeValues.MEDIUM}
          onlyIcon
        >
          <ArrowDownIcon />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test button component', () => {
  it('should be disabled (not active) ', () => {
    const { getByRole } = render(
      <Button
        variant={EButton.buttonVariantValues.PRIMARY}
        size={EButton.buttonSizeValues.MEDIUM}
        disabled
      />,
    );
    const button = getByRole('button');
    expect(button).toBeDisabled();
  });

  it('should run a function when clicked', () => {
    const mockFunction = jest.fn();
    const { getByText } = render(
      <Button
        variant={EButton.buttonVariantValues.PRIMARY}
        size={EButton.buttonSizeValues.MEDIUM}
        onClick={mockFunction}
      />,
    );
    const button = getByText('Button');
    userEvent.click(button);
    expect(mockFunction).toHaveBeenCalled();
  });
});
