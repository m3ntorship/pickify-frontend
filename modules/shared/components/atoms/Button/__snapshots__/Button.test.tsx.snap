// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render (disapled) button when we apply disapled 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-xs px-l text-base font-medium
       opacity-25 pointer-events-none"
  disabled={true}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (large) button when we apply (large) 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-m px-xl text-md font-bold
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (lefticon) button when we apply lefticon 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  Button
  <svg
    className="ml-xxs inline-block"
    fill="none"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
</button>
`;

exports[`Button component should render (meduim) button when we apply (meduim) 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (onlyicon)  button when we apply nothing 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       p-xs text-base font-medium
       "
  disabled={false}
  type="button"
/>
`;

exports[`Button component should render (primary) button when we apply primary 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (righticon) button when we apply righticon 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  <svg
    className="mr-xxs inline-block"
    fill="none"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
  Button
</button>
`;

exports[`Button component should render (secondary) button when we apply (secondary) 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       text-primary bg-white hover:bg-grey-bg  border border-primary focus:outline-none focus:ring-1 focus:ring-primary-dark focus:border-primary-dark 
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (small) button when we apply (small) 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-primary hover:bg-primary-hover text-white focus:outline-none focus:ring-1 focus:ring-primary-dark
       py-s px-m text-sm font-medium
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;

exports[`Button component should render (text) button when we apply text 1`] = `
<button
  className="rounded-full focus:outline-none ml-4 mt-4 focus:border-2 focus:border-dark-btnFocus

       bg-none text-accent hover:text-accent-hover focus:underline
       py-xs px-l text-base font-medium
       "
  disabled={false}
  type="button"
>
  Button
</button>
`;
